src/transformStateWithClones.js

```diff
@@ -1,13 +1,43 @@
 'use strict';
 
+const ACTION_ADD_PROPERTIES = 'addProperties';
+const ACTION_REMOVE_PROPERTIES = 'removeProperties';
+const ACTION_CLEAR_PROPERTIES = 'clear';
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+  let temporaryObject = { ...state };
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case ACTION_ADD_PROPERTIES:
+        temporaryObject = Object.assign(temporaryObject, extraData);
```

Comment:
```suggestion
        Object.assign(temporaryObject, extraData);
```

there is no need to set result of `Object.assign` to the `temporaryObject`

Reply:
Yes, there was a redundant piece of code